src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpUseImplicitTypingDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.Diagnostics.TypingStyles
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class CSharpUseImplicitTypingDiagnosticAnalyzer : CSharpTypingStyleDiagnosticAnalyzerBase
+    {
+
+        private static readonly LocalizableString s_Title =
+            new LocalizableResourceString(nameof(CSharpFeaturesResources.UseImplicitTypingDiagnosticTitle), CSharpFeaturesResources.ResourceManager, typeof(CSharpFeaturesResources));
+
+        private static readonly LocalizableString s_Message =
+            new LocalizableResourceString(nameof(CSharpFeaturesResources.UseImplicitTyping), CSharpFeaturesResources.ResourceManager, typeof(CSharpFeaturesResources));
+
+        public CSharpUseImplicitTypingDiagnosticAnalyzer()
+            : base(diagnosticId: IDEDiagnosticIds.UseImplicitTypingDiagnosticId,
+                   title: s_Title,
+                   message: s_Message)
+        {
+        }
+
+        protected override bool IsStylePreferred(SyntaxNode declarationStatement, SemanticModel semanticModel, OptionSet optionSet, State state, CancellationToken cancellationToken)
+        {
+            var stylePreferences = state.StylePreferences;
+            var shouldNotify = state.ShouldNotify();
+
+            // If notification preference is None, don't offer the suggestion.
+            if (!shouldNotify)
+            {
+                return false;
+            }
+
+            if (state.IsInIntrinsicTypeContext)
+            {
+                return stylePreferences.HasFlag(TypingStyles.VarForIntrinsic);
+            }
+            else if (state.IsTypingApparentInContext)
+            {
+                return stylePreferences.HasFlag(TypingStyles.VarWhereApparent);
+            }
+            else
+            {
+                return stylePreferences.HasFlag(TypingStyles.VarWherePossible);
+            }
+        }
+
+        protected override bool TryAnalyzeVariableDeclaration(TypeSyntax typeName, SemanticModel semanticModel, OptionSet optionSet, CancellationToken cancellationToken, out TextSpan issueSpan)
+        {
+            issueSpan = default(TextSpan);
```

Comment:
i don't like this.  it masks cases where you might miss a flow path.  Don't set here, but set later on when appropriate.